File: legacy/python-backend/app/websockets/market_stream.py
from app.websockets.binance_client import (
    BinanceWebSocketClient
)

from app.market.tick_processor import (
    TickProcessor
)

from app.candles.timeframe_candle_engine import (
    TimeframeCandleEngine
)

from app.core.logger import logger


class MarketStream:

    def __init__(self):

        self.client = BinanceWebSocketClient()

    async def start(self):

        logger.info(
            "Starting Market Stream"
        )

        async for tick in self.client.connect():

            await TickProcessor.process_tick(
                tick
            )

            await TimeframeCandleEngine.process_tick(
                symbol=tick["s"],
                price=float(tick["p"]),
                quantity=float(tick["q"])
            )

File: backend/src/websockets/marketStream.ts
import { BinanceWebSocketClient } from "./binanceClient";
import { TickProcessor } from "../trading/candles/tickProcessor";
import { TimeframeCandleEngine } from "../trading/candles/timeframeCandleEngine";
import { logger } from "../core/logger";

export class MarketStream {
  private client = new BinanceWebSocketClient();

  async start() {
    logger.info("Starting Market Stream");
    this.client.connect(async (tick) => {
      await TickProcessor.processTick(tick);
      await TimeframeCandleEngine.processTick(
        tick.s,
        parseFloat(tick.p),
        parseFloat(tick.q)
      );
    });
  }
}
export default MarketStream;
